archiver/stream_archiver.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package archiver
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+type StreamArchiver struct {
+	archPath string
+	*bytes.Reader
```

Comment:
Right, but take a look at `Archive(tw *tar.Writer)` function signature. It takes `Writter` which is the place where you are writing data to. You need also `Reader` to read it from in order to be able to `io.Copy` to `Writer`. So `Reader` is used here to copy data from stream to archive `tar.Writer`.